orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,482 @@
+import sys
+import copy
+import logging
+import concurrent.futures
+from functools import partial
+import time
+
+from AnyQt.QtWidgets import (
+    QApplication, QFormLayout, QTableView,  QSplitter)
+from AnyQt.QtCore import Qt, QThread, pyqtSlot, QMetaObject, Q_ARG
+import numpy as np
+from numpy.random import RandomState
+import scipy.stats as st
+
+import Orange
+import Orange. evaluation
+from Orange.widgets.widget import OWWidget, Output, Input, Msg
+from Orange.widgets import gui, widget, settings
+from Orange.widgets.utils.itemmodels import TableModel
+from Orange.widgets.utils.sql import check_sql_input
+from Orange.base import Model
+from Orange.data import DiscreteVariable, ContinuousVariable, StringVariable, Domain, Table
+from Orange.widgets.utils.concurrent import (
+    ThreadExecutor, FutureWatcher, methodinvoke
+)
+
+
+class Task:
+
+    future = ...
+    watcher = ...
+    canceled = False
+
+    def cancel(self):
+        self.canceled = True
+        self.future.cancel()
+        concurrent.futures.wait([self.future])
+
+
+class ExplainPredictions:
+    """
+    Class used to explain individual predictions by determining the importance of attribute values.
+    All interactions between atributes are accounted for by calculating Shapely value.
+
+    Parameters
+    ----------
+    data : Orange.data.Table
+        table with dataset
+    model: Orange.base.Model
+        model to be used for prediction
+    error: float
+        desired error 
+    p_val : float
+        p value of error
+    batch_size : int
+        size of batch to be used in making predictions, bigger batch size speeds up the calculations and improves estimations of variance
+    max_iter : int
+        maximum number of iterations per attribute
+    min_iter : int
+        minimum number of iterations per attiribute
+    seed : int
+        seed for the numpy.random generator, default is 667892
+
+    Returns:
+    -------
+    class_value: float
+        either index of predicted class or predicted value
+    table: Orange.data.Table
+        table containing atributes and corresponding contributions
+
+    """
+
+    def __init__(self, data, model, p_val=0.05, error=0.05, batch_size=500, max_iter=100000, min_iter=1000, seed=667892):
```

Comment:
Why is the default `seed=667892` ?
There are 2 standard options for seeds: 0 or 42 :)